assets/js/modules/reader-revenue-manager/components/setup/SetupMain.js

```diff
@@ -59,23 +60,29 @@ export default function SetupMain( { finishSetup = () => {} } ) {
 			SHOW_PUBLICATION_CREATE
 		)
 	);
-	const publicationID = useSelect( ( select ) =>
-		select( MODULES_READER_REVENUE_MANAGER ).getPublicationID()
+	const shouldResetPublications = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue(
+			READER_REVENUE_MANAGER_SETUP_FORM,
+			RESET_PUBLICATIONS
+		)
 	);
 
 	const { setValues } = useDispatch( CORE_FORMS );
 	const { resetPublications, submitChanges } = useDispatch(
 		MODULES_READER_REVENUE_MANAGER
 	);
 
-	const reset = useCallback( () => {
-		// Do not reset if the publication ID is already set.
-		if ( publicationID !== '' ) {
+	const reset = useCallback( async () => {
+		if ( ! shouldResetPublications ) {
```

Comment:
Since we're using a form key anyway, using it as part of publication create makes sense and is more concrete IMO, instead of just relying on no publication selected state.